src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,146 @@
+  return (
+    <li key={`${paginationUnitProps.keyIndex}-${paginationUnitProps.text}`}>
+      <button
+        data-testid={`pagination-button-page-${paginationUnitProps.keyIndex}`}
+        onClick={() => {
+          paginationUnitProps.onClick(paginationUnitProps.keyIndex);
+        }}
+        className={`pagination__button ${
+          paginationUnitProps.state === 'active' ? 'pagination__button--active' : ''
+        }`}
+      >
+        {paginationUnitProps.text}
+      </button>
+    </li>
+  );
+}
+
+function RenderPaginationBlock(paginationBlockProps: PaginationBlockProps) {
+  if (paginationBlockProps.totalPageCount === 1) {
+    const props = {
+      onClick: () => {
+        paginationBlockProps.onPageUnitClick(1);
+      },
+      state: 'active',
+      text: '1',
+      keyIndex: 1,
+    };
+
+    return <PaginationUnit {...props} />;
+  }
+
+  let paginationChildren: Array<any> = Array.from(
+    { length: paginationBlockProps.totalPageCount },
+    (_, i) => i + 1
+  );
+
+  paginationChildren = paginationChildren.map((page) => {
+    const props = {
+      onClick: () => {
+        paginationBlockProps.onPageUnitClick(page);
+      },
+      state: page === paginationBlockProps.currentPage ? 'active' : 'inactive',
+      text: page.toString(),
+      keyIndex: page,
+    };
+
+    return <PaginationUnit key={page} {...props} />;
+  });
+
+  paginationChildren.unshift(
+    PaginationUnit({
+      state: 'inactive',
+      text: 'Prev',
+      onClick: () => {
+        paginationBlockProps.onPreviousClick();
+      },
+      keyIndex: 0,
+    })
+  );
+
+  paginationChildren.push(
+    PaginationUnit({
+      state: 'inactive',
+      text: 'Next',
+      onClick: () => {
+        paginationBlockProps.onNextClick();
+      },
+      keyIndex: paginationBlockProps.totalPageCount + 1,
+    })
+  );
+
+  return paginationChildren;
+}
+
+export default function Pagination(paginationProps: PaginationProps) {
+  const { state, dispatch } = usePagination(paginationProps);
+
+  React.useEffect(() => {
+    paginationProps.handleChange(state.currentPage);
+  }, [state.currentPage]);
+
+  const currentPage = Number(state.currentPage);
+  const perPageLimit = Number(paginationProps.perPageLimit);
+
+  const startRowNumber = currentPage === 1 ? 1 : (currentPage - 1) * perPageLimit + 1;
+  const endRowNumber =
+    startRowNumber - 1 + perPageLimit > paginationProps.totalRowCount
+      ? paginationProps.totalRowCount
+      : startRowNumber - 1 + perPageLimit;
+
+  const onPreviousClick = () => {
+    dispatch({ type: PAGINATION_CHANGE.PREV_PAGE });
+  };
+
+  const onNextClick = () => {
+    dispatch({ type: PAGINATION_CHANGE.NEXT_PAGE });
+  };
+
+  const onPageUnitClick = (pageNumber: number) => {
+    dispatch({ type: PAGINATION_CHANGE.GO_TO_PAGE, payload: pageNumber });
+  };
+
+  if (!paginationProps.totalPageCount) return <></>;
+
+  return (
+    <nav className="pagination" data-testid="pagination">
```

Comment:
Avoid single-word class names: https://nimblehq.co/compass/development/code-conventions/css/#convention-24

```suggestion
    <nav className="pagination-navigation" data-testid="pagination">
```

Reply:
Fixed in https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/81/commits/9b7290de2b75163d97dca3a841796a4a5e735c95